lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -204,13 +200,15 @@ impl GelfDeserializer {
     }
 }
 
+#[serde_as]
 #[derive(Serialize, Deserialize, Debug)]
 struct GelfMessage {
     version: String,
     host: String,
     short_message: String,
     full_message: Option<String>,
-    timestamp: Option<f64>,
+    #[serde_as(as = "Option<TimestampSecondsWithFrac<f64>>")]
+    timestamp: Option<DateTime<Utc>>,
```

Comment:
I brought in `serde_with` for custom deserialization rather than doing floating point arithmetic to preserve the precision of the timestamps.